src/components/BubbleSort.js

```diff
@@ -0,0 +1,79 @@
+export default class BubbleSort {
+  constructor(elements) {
+    this.numberElements = elements;
+  }
+
+  sortingElements() {
+    const elements = this.numberElements;
+    const l = elements.length;
+    const historyElements = [[...elements]];
+    const animations = [];
```

Comment:
Należałoby zrobić klasę AnimationsBuilder którą przekażesz przez dependency injection

`class AnimationsBuilder {

constructor() {
this.animations = []
}
function add(animation) {
this.animations.push(animation)
}

function getAll() {
return this.animations
}
}`


`
const animationsBuilder = new AnimationsBuilder()`
const bubbleSort = new BubbleSort(animationsBuilder)
bubbleSort.sort()
const animations = animationsBuilder.getAll()
`

Reply:
Co z tym?